extensions/stats/plugin.cc

```diff
@@ -460,8 +462,14 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
     // For TCP, if peer metadata is not available, peer id is set as not found.
     // Otherwise, we wait for metadata exchange to happen before we report  any
     // metric.
+    // We skip this wait  if upstream is unhealthy as we won't get peer metadata
+    // in that case.
+    uint64_t response_flags = 0;
+    getValue({"response", "flags"}, &response_flags);
     if (peer_node_ptr == nullptr &&
-        peer_id != ::Wasm::Common::kMetadataNotFoundValue) {
+        peer_id != ::Wasm::Common::kMetadataNotFoundValue &&
+        !(::Wasm::Common::parseResponseFlag(response_flags)
```

Comment:
Response flags is a bit mask so just use XOR to check for a flag.

Reply:
Fixed